File: app/src/main/java/org/stepik/android/adaptive/pdd/ui/dialog/LogoutDialog.java
package org.stepik.android.adaptive.pdd.ui.dialog;

import android.app.Dialog;
import android.content.DialogInterface;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.v4.app.DialogFragment;
import android.support.v7.app.AlertDialog;

import org.stepik.android.adaptive.pdd.R;
import org.stepik.android.adaptive.pdd.core.LogoutHelper;
import org.stepik.android.adaptive.pdd.core.ScreenManager;

public final class LogoutDialog extends DialogFragment implements Dialog.OnClickListener {

    @NonNull
    @Override
    public Dialog onCreateDialog(final Bundle savedInstanceState) {
        final AlertDialog.Builder alertDialogBuilder = new AlertDialog.Builder(getContext());
        alertDialogBuilder.setTitle(R.string.logout);
        alertDialogBuilder.setMessage(R.string.logout_dialog);
        alertDialogBuilder.setPositiveButton(android.R.string.yes, this);
        alertDialogBuilder.setNegativeButton(android.R.string.no, this);

        return alertDialogBuilder.create();
    }

    @Override
    public void onClick(DialogInterface dialogInterface, int which) {
        if (which == DialogInterface.BUTTON_POSITIVE) {
            LogoutHelper.logout(ScreenManager.getInstance()::showLaunchScreen);
        }
    }
}

File: app/src/main/java/org/stepik/android/adaptive/ui/dialog/LogoutDialog.kt
package org.stepik.android.adaptive.ui.dialog

import android.app.Dialog
import android.content.DialogInterface
import android.os.Bundle
import androidx.appcompat.app.AlertDialog
import androidx.fragment.app.DialogFragment
import org.stepik.android.adaptive.App
import org.stepik.android.adaptive.R
import org.stepik.android.adaptive.core.LogoutHelper
import org.stepik.android.adaptive.core.ScreenManager
import javax.inject.Inject

class LogoutDialog : DialogFragment(), DialogInterface.OnClickListener {
    @Inject
    lateinit var logoutHelper: LogoutHelper

    @Inject
    lateinit var screenManager: ScreenManager

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        App.component().inject(this)
    }

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val alertDialogBuilder = AlertDialog.Builder(requireContext())
        alertDialogBuilder.setTitle(R.string.logout)
        alertDialogBuilder.setMessage(R.string.logout_dialog)
        alertDialogBuilder.setPositiveButton(android.R.string.yes, this)
        alertDialogBuilder.setNegativeButton(android.R.string.no, this)

        return alertDialogBuilder.create()
    }

    override fun onClick(dialogInterface: DialogInterface, which: Int) {
        if (which == DialogInterface.BUTTON_POSITIVE) {
            logoutHelper.logout(screenManager::showOnboardingScreen)
        }
    }
}
